### PR TITLE
Mayank/namespace auto proxy

### PR DESCRIPTION
**Pull request by mayankshah1607**
Fixes issue #3255

### CODE REVIEW

File: cli/cmd/inject.go [hunk 1]
```diff
@@ -156,6 +162,7 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 		conf.AppendPodAnnotations(rt.overrideAnnotations)
 	}
 
+	// Need help from this part
```
Comment (mayankshah1607) [inline cli/cmd/inject.go]: I need help starting from this part. The annotations are being injected to `conf.workload.Meta.Annotations`. However, the generated JSON is still the same as the input.

File: cli/cmd/inject.go [hunk 2]
```diff
@@ -165,6 +172,8 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 	}
 	log.Infof("patch generated for: %s", report.ResName())
 	log.Debugf("patch: %s", patchJSON)
+
```
Comment (mayankshah1607) [inline cli/cmd/inject.go]: Follow up to the previous comment. The patch generated is an empty array. Should this be of any concern? Am I missing something?

**Comment by ihcsim** [comment]
@mayankshah1607 @Pothulapati Sorry for the delay. I'll have time to review this either later this week or early next. Meanwhile, can you merge this with `master`, and re-run the unit test and lint to make sure everything is still :heavy_check_mark:? (The unit test failure might go over when you push the new commits to this branch.) Thanks!

**Comment by Pothulapati** [comment]
Hey, 
Sorry for a bit late follow up.

I pulled the PR but had to go through a different route, to make annotations come through patching. 

By having these annotations, being added on the patch side, commands with flags i.e 
```
linkerd inject --proxy-memory-request 200m ./file.yaml
```
will add the necessary config annotations, along with the inject annotaiton to the namespace.

You can check my code changes here https://github.com/Pothulapati/linkerd2/commit/1e08725d8e0fe37c1c2fc072803e0535fb28fa33

Can you follow that PR and make necessary changes? 

Once you are done with that,

- Update the `linkerd install` golden files, as they also go through the `transform` process (to make the control plane components also have proxies) by running
`env GO111MODULE=on go test ./cli/cmd/... -v --update`
- Try writing additional tests, by having additional `overrideAnnotations` in `inject_test.go` . This will make the unit tests pass.
- Make sure there are no linting errors by checking `env GO111MODULE=on ./bin/lint --verbose`

**Comment by mayankshah1607** [comment]
Hi, I am closing this PR and opening a cleaner version of the same so that it is easier to review - #3549

**Comment by ihcsim** [comment]
@mayankshah1607 Thanks :+1:!

**Pull request closed without merging**